consensus/raft/consensus.go

```diff
@@ -293,17 +305,12 @@ func (cc *Consensus) LogUnpin(c api.Pin) error {
 // forward the operation to the leader if this is not it.
 func (cc *Consensus) LogAddPeer(addr ma.Multiaddr) error {
 	var finalErr error
-	for i := 0; i < CommitRetries; i++ {
+	for i := 0; i <= cc.config.CommitRetries; i++ {
 		logger.Debugf("Try %d", i)
 		redirected, err := cc.redirectToLeader(
```

Comment:
The structure of the code is definitely pretty tricky around here.  It took a while to come up with an explanation for why you are adding the retry loop in the redirect function (as far as I can tell it is to enforce that redirects also get the right number of retries).  It might be worth thinking about how to refactor `redirectToLeader` so that all of the looping can go on outside of it and we only need to worry about retries in one place.  Also I have a natural inclination to play around with the three functions LogRmPeer, LogAddPeer and LogOpCid to see if we can abstract away most of the redirection and retrying functionality to one function.

Reply:
Actual the reason I added retry loop in the `redirectToLeader` was an attempt to simplify the code. Before, if the redirectRPC failed, it consumed one retry and retry immediately. I discover a case where the leader was just gone, but all retries would be consumed without even realizing it was gone. Therefore I needed to add a time.Sleep() for it (x3, because there are 3 functions doing the same). And in the end, this was retry logic and delays for failures in `redirectToLeader`, so I figured it was just better to deal with that part in a single place directly in the function.

> Also I have a natural inclination to play around with the three functions LogRmPeer, LogAddPeer and LogOpCid to see if we can abstract away most of the redirection and retrying functionality to one function.

I'm going to give it another round. There has to be a way to make things cleaner.